packages/block-library/src/block/edit.js

```diff
@@ -38,6 +41,23 @@ import { unlock } from '../lock-unlock';
 const { useLayoutClasses } = unlock( blockEditorPrivateApis );
 const fullAlignments = [ 'full', 'wide', 'left', 'right' ];
 
+function isPartiallySynced( block ) {
+	return (
+		!! block.attributes.connections?.attributes &&
+		Object.values( block.attributes.connections.attributes ).some(
+			( connection ) => connection.source === 'pattern_attributes'
+		)
+	);
+}
+
+function setBlockEditMode( setEditMode, block ) {
```

Comment:
Nit: We can accept `blocks` array here so that we can just call `setBlockEditMode( setEditMode, blocks )` in the effect.

Reply:
nice, that is much tidier, done.